ejb3/src/main/java/org/jboss/as/ejb3/component/singleton/SingletonComponentDescription.java

```diff
@@ -125,7 +125,9 @@ public void configure(final DeploymentPhaseContext context, final ComponentDescr
                     if(interceptorConfig.getPostConstruct() != null) {
```

Comment:
@dmlloyd David, according to EJB 3.2 spec section 4.3.4, Singletons and Stateful Bean are the only beans with defined tx context in lifecycle callback interceptor methods. Regarding Stateless and MDBs (section 5.4.6 of the spec) it should be safe to remove these hooks and thus remove redundant no-op txs from them.

But you are right that the `interceptorConfig` does not reflect any `@Interceptors` annotated on the bean. I've checked it. So we cannot easily remove it from Singleton and Stateful beans as it has already happened on line 125 in the past. At least, this has to be changed.